### PR TITLE
Add getNotificationById

### PR DESCRIPTION
**Pull request by andrewoh531**

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -221,3 +229,22 @@ type TemplatePreview struct {
...
+type Notification struct {
+	ID           string `json:"id,omitempty"`
+	Subject      string `json:"subject"`
+	Body         string `json:"body"`
+	Reference    string `json:"reference"`
+	EmailAddress string `json:"email_address"`
+	PhoneNumber  string `json:"phone_number"`
+	Type         string `json:"type"`
+	Status       string `json:"status"`
+	CreatedBy    string `json:"created_by_name"`
```
Comment (jonathaningram) [inline client.go:242]: @andrewoh531 `created_by_name`?

Comment (dagoof) [reply client.go:242]: We don't have any API around users. It would be useless to return a userID -- I assume that's why this gives back a name

**Review by dagoof — APPROVED** [review]

File: client.go (hunk 1 above)
Comment (jonathaningram) [reply client.go:242]: I searched for `created_by_name` in the codebase and got 0 results. Did I do a bad search?

Comment (andrewoh531) [reply client.go:242]: It's in the client-sdk for python, node

Comment (dagoof) [reply client.go:242]: https://github.com/govau/notify-client-dotnet/blob/1ef5ff29fd52836b3896d24b11ef8f3a1a943d9d/src/Notify/Models/Notification.cs#L38

Comment (jonathaningram) [reply client.go:242]: We worked it out offline. We haven't pulled in this field from upstream yet.

**Pull request merged**